src/main/java/MeggyTest.java

```diff
@@ -0,0 +1,89 @@
+import Meggy.Exception.MeggyException;
+import Meggy.Meggy;
+import Meggy.MeggyTime;
+import Meggy.Task.DdlTask;
+import Meggy.Task.EventTask;
+import Meggy.Task.TodoTask;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.Random;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * For testing purpose only.
+ */
+public class MeggyTest {
+    private static final Random RAND = new Random();
+    private static final int N_TEST = 1024;
+
+    /**
+     * @return String that will never be entirely whitespace.
+     */
+    private static String randString() {
+        while (true) {
+            final int n = 1 + RAND.nextInt(50);
+            char[] s = new char[n];
+            boolean allSpace = true;
+            for (int i = 0; i < n; i++) {
+                final char c = (char) (' ' + RAND.nextInt(95));
+                s[i] = c;
+                allSpace &= c == ' ';
+            }
+            if (!allSpace) {
+                return new String(s);
+            }
+        }
+    }
+
+    private static MeggyTime randMeggyTime() {
+        return MeggyTime.of(RAND.nextDouble() < 0.5 ? randString()
+                : LocalDateTime.ofEpochSecond(RAND.nextInt(), 0, ZoneOffset.UTC).format(MeggyTime.ENCODE_FORMAT));
```

Comment:
Maybe try to write ternary expressions in this form?
```suggestion
        return MeggyTime.of(RAND.nextDouble() < 0.5 
                ? randString()
                : LocalDateTime.ofEpochSecond(RAND.nextInt(), 0, ZoneOffset.UTC).format(MeggyTime.ENCODE_FORMAT));
```